packages/popover/src/index.tsx

```diff
@@ -102,6 +102,35 @@ const usePopover = ({
     []
   );
 
+  const preventOverflow = useCallback(
+    (
+      popoverNode: HTMLElement,
+      position: Pick<DOMRect, PlacementGetterType>
+    ) => {
+      const { left } = position;
+      const { scrollWidth } = popoverNode;
+      const documentBodyWidth = document.body.scrollWidth;
+      const popoverNodeStyles = window.getComputedStyle(popoverNode);
+      const { borderLeftWidth, borderRightWidth } = popoverNodeStyles;
+
+      const borderWidth =
+        parseFloat(borderLeftWidth) + parseFloat(borderRightWidth);
+
+      const horizontalOverflowAmount =
+        left + scrollWidth + borderWidth - documentBodyWidth;
+
+      const isHorizontalyOverflowed = horizontalOverflowAmount > 0;
+      const hasEnoughSpaceOnLeft = left - horizontalOverflowAmount >= 0;
+
+      if (isHorizontalyOverflowed && hasEnoughSpaceOnLeft) {
```

Comment:
Hey, @oak93! There is a question. If it is horizontalyOverflowed and it is not enoughSpaceOnLeft, the popover will be rendered as overflowed. It should be rendered in this case? It could still seem buggy. But I couldn't figure out a proper way too :)

Reply:
I believe the design of the popover's content should change in that situation. 
the width of the popover should not be bigger than the size of the clientWidth of body. @Ogzhnsfgl 
I checked the behavior of the Popperjs to get more understanding of this situation. it is also the same there. you can check here my changing the text of Popperjs example on Codesandox https://popper.js.org/